### PR TITLE
Remove Registry

### PR DESCRIPTION
**Pull request by nouseforaname**
### What is this change about?

Remove Registry from codebase & manifests. It should not be used anymore

### Please provide contextual information.
https://www.pivotaltracker.com/story/show/176871612

### What tests have you run against this PR?

`bundle exec rake spec:unit`
```
Finished in 16 minutes 1 second (files took 6.68 seconds to load)
7311 examples, 0 failures, 12 pending```
```
`fly -t bosh-ecosystem execute -j bosh-director/integration-db-tls-postgres -c ci/tasks/test-integration.yml -i bosh-src=../bosh -p`
```
1154 examples, 0 failures

Took 3166 seconds (52:46)
```

### How should this change be described in bosh release notes?

Remove Registry Code & Deployment options. This will drop support for trusty?

### Does this PR introduce a breaking change?

No more support for V1 Stemcells  probably

### CODE REVIEW

File: src/bosh-director/spec/unit/problem_handlers/unresponsive_agent_spec.rb [hunk 1]
```diff
@@ -1,346 +0,0 @@
-require 'spec_helper'
```
Comment (beyhan) [inline src/bosh-director/spec/unit/problem_handlers/unresponsive_agent_spec.rb:1]: I'm wandering why this change deletes the src/bosh-director/spec/unit/problem_handlers/unresponsive_agent_spec.rb?

**Comment by lnguyen** [comment]
Waiting on verification of different CPI to see if they still use registry

File: src/bosh-director/spec/unit/problem_handlers/unresponsive_agent_spec.rb (hunk 1 above)
Comment (nouseforaname) [reply src/bosh-director/spec/unit/problem_handlers/unresponsive_agent_spec.rb]: you are totally right (again). Removing that file looks like an accident and was reverted.

**Review by beyhan — APPROVED** [review]

File: src/bosh-director/spec/unit/problem_handlers/unresponsive_agent_spec.rb (hunk 1 above)
Comment (beyhan) [reply src/bosh-director/spec/unit/problem_handlers/unresponsive_agent_spec.rb]: thanks :-)

**Comment by lnguyen** [comment]
Should we be removing all the gems? @nouseforaname

**Comment by rkoster** [comment]
@lnguyen Softlayer has been notified and they are aware that they will be stuck on the last director version which still supports the registry (to be determined). Once this PR gets a second approval it can be merged.

**Comment by lnguyen** [comment]
Cool, I just have a few questions for @nouseforaname on PR

**Review by lnguyen — APPROVED** [review]

**Comment by rkoster** [comment]
@lnguyen would you care to share your questions?

**Comment by lnguyen** [comment]
oops... i realized I never clicked `submit review` but I've since talked to @nouseforaname it was rubocop changes that I didn't realize

**Comment by bgandon** [comment]
First of all, I'm glad that the Bosh codebase evolves and that we deprecate old things and move on—about that, shouldn't we speak about “_Bosh v3_” like we had “_Bosh v2_” feature set few years ago?

For this PR specifically, Konstantin, could you use `git rebase -i` and merge your latest commit c4c0cd5 with previous ones, so that the history of those files is better preserved?

I've looked for remaining occurrences of `registry` in the code, and found those three concerns:

- Are the following comments in `Gemfile` related to the registry that is being removed? Are they still relevant?
https://github.com/cloudfoundry/bosh/blob/feature%2F176871612-remove-registry/src/Gemfile#L23-L24

- What about this comment from 608affe88: https://github.com/cloudfoundry/bosh/commit/608affe8853ed9090701041dcb16140a40995ed1#diff-a044a5365a1ddf0d279a38eb1cc4b6e79fa15fa2a1f28f1d804fd0d8744c7727R182
Is it still relevant?

- What about this comment from 3bdf5e2961: https://github.com/cloudfoundry/bosh/commit/3bdf5e29615fa9874b8c65f7a20a06e75116498e#diff-a29a44ce81d812a9add92faa832151f8440e4046ede52391d6fe85db2d7ad599R53
Shouldn't it be removed?

Furthermore, it would be nice that we share plans for consequences of this removal and sunset of CPI v1 API in general:
- complete removal of CPI v1 support in Director's codebase,
- deadlines for sunset of CPI v1 support in Bosh-Agent and Stemcells
- sunset of CPI v1 API support in official CPIs codebase
- proper communication to CPI authors about this sunset of v1 API and actions to take
- related updates in bosh.io documentation

Some Google Docs maybe?

**Comment by nouseforaname** [comment]
> First of all, I'm glad that the Bosh codebase evolves and that we deprecate old things and move on—about that, shouldn't we speak about “_Bosh v3_” like we had “_Bosh v2_” feature set few years ago?
> 
> For this PR specifically, Konstantin, could you use `git rebase -i` and merge your latest commit [c4c0cd5](https://github.com/cloudfoundry/bosh/commit/c4c0cd58c76219dcd16ab7ce15f7782a92001f93) with previous ones, so that the history of those files is better preserved?
> 
> I've looked for remaining occurrences of `registry` in the code, and found those three concerns:
> 
>     * Are the following comments in `Gemfile` related to the registry that is being removed? Are they still relevant?
>       https://github.com/cloudfoundry/bosh/blob/feature%2F176871612-remove-registry/src/Gemfile#L23-L24
> 
>     * What about this comment from [608affe](https://github.com/cloudfoundry/bosh/commit/608affe8853ed9090701041dcb16140a40995ed1): [608affe#diff-a044a5365a1ddf0d279a38eb1cc4b6e79fa15fa2a1f28f1d804fd0d8744c7727R182](https://github.com/cloudfoundry/bosh/commit/608affe8853ed9090701041dcb16140a40995ed1#diff-a044a5365a1ddf0d279a38eb1cc4b6e79fa15fa2a1f28f1d804fd0d8744c7727R182)
>       Is it still relevant?
> 
>     * What about this comment from [3bdf5e2](https://github.com/cloudfoundry/bosh/commit/3bdf5e29615fa9874b8c65f7a20a06e75116498e): [3bdf5e2#diff-a29a44ce81d812a9add92faa832151f8440e4046ede52391d6fe85db2d7ad599R53](https://github.com/cloudfoundry/bosh/commit/3bdf5e29615fa9874b8c65f7a20a06e75116498e#diff-a29a44ce81d812a9add92faa832151f8440e4046ede52391d6fe85db2d7ad599R53)
>       Shouldn't it be removed?
> 
> 
> Furthermore, it would be nice that we share plans for consequences of this removal and sunset of CPI v1 API in general:
> 
>     * complete removal of CPI v1 support in Director's codebase,
> 
>     * deadlines for sunset of CPI v1 support in Bosh-Agent and Stemcells
> 
>     * sunset of CPI v1 API support in official CPIs codebase
> 
>     * proper communication to CPI authors about this sunset of v1 API and actions to take
> 
>     * related updates in bosh.io documentation
> 
> 
> Some Google Docs maybe?

Hey, thanks for checking the PR. I removed the comments and rebased the commits (it took me a few attempts :/).
The comments are gone and as far as I understand the commits that introduced them, they relate to introducing a wrapper for the dummy at some point. 

As far as your docs idea goes, I'm not quite sure how to provide that? I'm still quite new to the ecosystem work and I do not know who to involve and where to start.

**Comment by rkoster** [comment]
@bgandon With regards to the further deprecation of amongst other things the v1 CPI, I would suggest creating an RFC for that. This is also currently being done for the [jammy jellyfish adoption](https://github.com/cloudfoundry/community/pull/191). The RFC process is described [here](https://github.com/cloudfoundry/community/pull/174)

**Comment by jpalermo** [comment]
> Furthermore, it would be nice that we share plans for consequences of this removal and sunset of CPI v1 API

Opened a new discussion here: https://github.com/cloudfoundry/bosh/discussions/2355

**Comment by rkoster** [comment]
Thanks! @nouseforaname
